### ISSUE TITLE
Can't highlight router links in parent layout anymore

### ISSUE THREAD

**Issue by heruan** [vaadin/flow#3687]
After #3636 if a parent layout has links which should react on navigation events (e.g. highlight), they don't anymore. Use case:

```java
public class SectionLayout extends Div implements RouterLayout {

    private final Div navbar = new Div();

    private final Div container = new Div();

    public SectionLayout() {
        RouterLink fooLink = new RouterLink("Foo", Foo.class);
        RouterLink barLink = new RouterLink("Bar", Bar.class);
        fooLink.setHighlightAction(HighlightActions.toggleAttribute("highlight");
        barLink.setHighlightAction(HighlightActions.toggleAttribute("highlight");
        navbar.add(fooLink, barLink);
        add(navbar, container);
    }

    @Override
    void setRouterLayoutContent(HasElement content) {
        container.getElement().appendChild(content.getElement());
    }

}

@Route(value = "foo", layout = SectionLayout.class)
public class Foo extends Div {}

@Route(value = "bar", layout = SectionLayout.class)
public class Bar extends Div {}
```

The two links now don't highlight after navigation events, but they did before #3636 and #3629.

I would expect any component in the DOM to react to navigation events if it implements the proper interfaces. Another use case: a widget component in the parent layout, which changes its content based on navigation events.

**Comment by Legioth** [comment]
Related to https://github.com/vaadin/flow/issues/3649

**Comment by heruan** [comment]
Let me understand better. In #3649 you state:

> `AfterNavigationObserver` should be notified once for any component that is actually attached after applying the navigation state.
> - The main use case is to allow updates to e.g. breadcrumbs or the active state of router links.

But that is exactly what #3636 prevented. What am I missing?

**Comment by Legioth** [comment]
#3636 was the result of a misunderstanding.

**Comment by pleku** [comment]
This should be fixed by #3649, but we should verify that and add test for the use case mentioned here.
